ui/org.eclipse.pde.bnd.ui/src/org/eclipse/pde/bnd/ui/internal/SelectionUtils.java

```diff
@@ -0,0 +1,63 @@
+package org.eclipse.pde.bnd.ui.internal;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import org.eclipse.core.runtime.IAdaptable;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.osgi.util.function.Predicate;
+
+public class SelectionUtils {
+
+	public static <T> Collection<T> getSelectionMembers(ISelection selection, Class<T> clazz) throws Exception {
+		return getSelectionMembers(selection, clazz, null);
+	}
+
+	public static <T> T adaptObject(Object obj, Class<T> clazz) {
+		if (clazz.isInstance(obj)) {
+			@SuppressWarnings("unchecked")
+			T result = (T) obj;
+			return result;
+		}
+
+		if (obj instanceof IAdaptable) {
+			T result = ((IAdaptable) obj).getAdapter(clazz);
+			return result;
+		}
+
+		return null;
+	}
+
+	public static <T> Collection<T> getSelectionMembers(ISelection selection, Class<T> clazz,
+		Predicate<? super T> filter) throws Exception {
+		if (selection.isEmpty() || !(selection instanceof IStructuredSelection)) {
+			return Collections.emptyList();
+		}
+
+		IStructuredSelection structSel = (IStructuredSelection) selection;
+		List<T> result = new ArrayList<>(structSel.size());
+		Iterator<?> iter = structSel.iterator();
+		while (iter.hasNext()) {
```

Comment:
Strucutred selection has a stream() method that seems better used here, also as above `Adapters.of(...).stream()` can be used